hre/src/main/java/hre/config/Configuration.java

```diff
@@ -297,6 +303,53 @@ public static MessageProcessEnvironment getSilicon() throws IOException {
         return env;
     }
 
+    public static String getMonoVersion() {
+        MessageProcessEnvironment env = new MessageProcessEnvironment("mono");
+        env.addArg("--version");
+        MessageProcess mp = env.startProcess();
+        while (!mp.isFinished()) {
+            Output("Waiting for termination of \"mono --version\"...");
```

Comment:
uh, what's up with this busy-waiting loop?

Reply:
To get the output of "mono --version", we have to wait. Not sure if this can be done asynchronously, as the whole execution of verification and the carbon backend depends on this info. I can insert a sleep(1ms) to make the busy wait less cpu-intensive.

Reply:
I was indeed less worried about the (synchronous) wait-part, and more the busy-part :). You should be able to make a method that uses Process.waitFor in MessageProcess.

Reply:
I refactored it to use raw java processes, using "waitFor" is more efficient and more readable.